core/trino-main/src/main/java/io/trino/server/ui/MultipartUiCookie.java

```diff
@@ -0,0 +1,128 @@
+import static java.util.Objects.requireNonNull;
+
+public class MultipartUiCookie
+{
+    // https://chromestatus.com/feature/4946713618939904
+    private static final int COOKIE_LENGTH_LIMIT = 4096;
+
+    private final String cookieName;
+    private final String location;
+
+    public MultipartUiCookie(String cookieName, String location)
+    {
+        this.cookieName = requireNonNull(cookieName, "cookieName is null");
+        this.location = requireNonNull(location, "location is null");
+    }
+
+    public NewCookie[] create(String token, Instant tokenExpiration, boolean isSecure)
+    {
+        Date expiration = Optional.ofNullable(tokenExpiration).map(Date::from).orElse(null);
+        ImmutableList.Builder<NewCookie> cookiesToSet = ImmutableList.builder();
+        int index = 0;
+        for (String part : splitValueByLength(token)) {
+            cookiesToSet.add(new NewCookie.Builder(cookieName(index++))
+                    .value(part)
+                    .path(location)
+                    .domain(null)
+                    .comment(null)
+                    .maxAge(DEFAULT_MAX_AGE)
+                    .expiry(expiration)
+                    .secure(isSecure)
+                    .httpOnly(true)
+                    .build());
+        }
+        return cookiesToSet.build().toArray(new NewCookie[0]);
+    }
+
+    public Optional<String> read(Map<String, Cookie> existingCookies)
+    {
+        long cookiesCount = existingCookies.values().stream()
+                .filter(cookie -> cookie.getName().startsWith(cookieName))
+                .filter(cookie -> !isNullOrEmpty(cookie.getValue()))
+                .count();
+
+        if (cookiesCount == 0) {
+            return Optional.empty();
+        }
+
+        StringBuilder token = new StringBuilder();
+        for (int i = 0; i < cookiesCount; i++) {
+            Cookie cookie = existingCookies.get(cookieName(i));
+            if (cookie == null || isNullOrEmpty(cookie.getValue())) {
+                return Optional.empty(); // non continuous
+            }
+            token.append(cookie.getValue());
+        }
+        return Optional.of(token.toString());
+    }
+
+    public NewCookie[] delete(Map<String, Cookie> existingCookies, boolean isSecured)
+    {
+        ImmutableSet.Builder<NewCookie> cookiesToDelete = ImmutableSet.builder();
+        cookiesToDelete.add(deleteCookie(cookieName, isSecured)); // Always add first cookie even if it doesn't exist
+        for (Cookie existingCookie : existingCookies.values()) {
+            if (existingCookie.getName().startsWith(cookieName)) {
+                cookiesToDelete.add(deleteCookie(existingCookie.getName(), isSecured));
+            }
+        }
+        return cookiesToDelete.build().toArray(new NewCookie[0]);
+    }
+
+    private static List<String> splitValueByLength(String value)
+    {
+        return Splitter.fixedLength(COOKIE_LENGTH_LIMIT).splitToList(value);
```

Comment:
It's 4096 for both, the name and the value:
> After the spec change corresponding to this Intent, user agents are now required to limit the **sum of the lengths** of the cookie's name and value to 4096 bytes

Reply:
oh, that's important then